### DISCUSSION TITLE
Cross Probing Multiple Screens

### DISCUSSION THREAD

**Discussion by Rycha2021** [iTwin/itwinjs-core#3608] [Q&A] [answered]
Hi iTwin team,

We are considering on embarking on a path to develop Cross Probing with multiple views. We are considering 4 views that the Client would like to be synchronised should the user require.

However, in various discussions offline between developers, there was a view that anymore than 2 views may not be possible with iTwin's libraries.
 
Can we please receive guidance if this view is in fact correct, i.e there are / there are not limitations to have 4 views? 

We hope to have a definitive answer before embarking on the development activity.

Can you please point us in the right direction of any existing code start for the same?

Thanks. Ry

**Comment by pmconne** [comment]
Where did you get the impression that iTwin.js won't support more than 2 views?
You can open any number of viewports in any arrangement. All the viewport requires is a `div` for it to inhabit. For example here are 4 independently-positioned viewports in one of our test apps:
![image](https://user-images.githubusercontent.com/22944042/168066631-dda078bb-6530-4fbb-9eae-fda4ae58de63.png)

If you're using the iTwin.js UI libraries:
[Four viewports in a grid](https://github.com/iTwin/itwinjs-core/discussions/3553#discussioncomment-2638610)
[One viewport floating atop another viewport](https://www.itwinjs.org/learning/ui/appui-react/contentviews/#floating-content-view).

**Comment by Rycha2021** [comment]
Thanks for the rapid response @pmconne.

Long story for the background why the query was raised which i wont get into here.

Interesting this was also a recent query. My miss.

Notwithstanding, "**synchronised**" is the key word here not solely 4 windows.

"TwoWayViewportSync could be generalized and replaced with a MultiViewportSync API accepting a list of 2 or more viewports. If you have success, that would be a nice PR." hence appears this is unproven and will require effort to resolve.....

Guidance welcomed!

Thanks Ry.

**Comment by pmconne** [comment]
> appears this is unproven and will require effort to resolve.....

It requires almost no effort to resolve, that's why I suggested it would make for a nice PR. I guess I'll submit one.

**Reply by Rycha2021** [reply]
That will be great @pmconne. We will look our for the PR. 
Your response was very much appreciated and clarified our misunderstanding.
Ry.

**Comment by pmconne** [comment, marked as answer]
See #3614

**Reply by pmconne** [reply]
This feature will be included in the imminent 3.2.0 release.

**Reply by Rycha2021** [reply]
Thanks Paul.
Ry
